src/restconfig/src/main/java/org/geoserver/rest/security/xml/AuthProvider.java

```diff
@@ -0,0 +1,87 @@
+/* (c) 2025 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.rest.security.xml;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import org.geoserver.security.config.SecurityAuthProviderConfig;
+
+@XStreamAlias("authProvider")
+public class AuthProvider {
+    private String id;
+    private String name;
+    private String className;
+    private String userGroupServiceName;
+    private int position;
+    private SecurityAuthProviderConfig config;
+
+    // Available & Selected, disabled when available but not selected.
+    private boolean disabled;
+
+    public AuthProvider() {}
+
+    public AuthProvider(SecurityAuthProviderConfig provider) {
+        this.id = provider.getId();
+        this.name = provider.getName();
+        this.className = provider.getClassName();
+        this.userGroupServiceName = "default";
+        this.config = provider;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getClassName() {
+        return className;
+    }
+
+    public void setClassName(String className) {
+        this.className = className;
+    }
+
+    public boolean isDisabled() {
```

Comment:
How does one get to have a disabled provider in the GeoServer auth providers configuration?

![image](https://github.com/user-attachments/assets/fc755b48-6ef2-4d63-9f21-94da0a54d416)

The main configuration only has a list of names:

https://github.com/geoserver/geoserver/blob/main/data/release/security/config.xml#L3

and the single provider configuration does not have a notion of enabled/disabled:

https://github.com/geoserver/geoserver/blob/main/data/release/security/auth/default/config.xml

Seems to me that a provider is there, or not, but cannot be there and be disabled?